core/core.pro

```diff
@@ -17,8 +17,7 @@ INCLUDEPATH += $$SENSORFW_INCLUDEPATHS
 
 DEFINES += PLUGIN_DIRECTORY=\"\\\"$$[QT_INSTALL_LIBS]/sensord-qt5\\\"\"
 
-QMAKE_LIBDIR_FLAGS += -L../datatypes\
-                      -lsensordatatypes-qt5
+LIBS += -L$$[QT_INSTALL_LIBS] -L../datatypes -lsensordatatypes-qt5
```

Comment:
This doesn't look right. The datatypes should be linked against the built version, not the installed one. And the root .pro file anyway has "CONFIG += ordered" and datatypes before the core so the library should be there?

Reply:
See https://ci.ubports.com/blue/organizations/jenkins/UBportsCore%2FPackaging%20%252F%20sensorfw/detail/MR-12/2/pipeline

Reply:
On hybris the current form with -L$$[QT_etc coming from commit 1dd09c7b91c. Earlier it was linking against hybris libs so it made some sense, but on that commit I suppose it might be a leftover and the local directories could suffice.

cc @mlehtima ^

Regardless I'm not following how this change would fix your build since the LIBS shouldn't affect the order of build nor dependencies. Maybe works by accident if using an earlier installed library from the system(?) 

Also puzzled why the ordered build is not honored on your build. But if it helps, it could be nice to have that replaced with subdirectory dependencies a la https://wiki.qt.io/SUBDIRS_-_handling_dependencies

Reply:
This discussion is obsolete as the patch has entirely changed.